### ISSUE TITLE
Support for Phoenix 1.7

### ISSUE THREAD

**Issue by kyleboe** [thoughtbot/bamboo_phoenix#7]
With Phoenix 1.7 replacing `Phoenix.View` with `Phoenix.Component` (see [docs in `:phoenix_view`](https://hexdocs.pm/phoenix_view/Phoenix.View.html#module-replaced-by-phoenix-component)), `bamboo_phoenix` will need to be updated. This update will likely need to involve a few steps: 

1. Setting an upper limit on the currently specified phoenix version (involves a version release i.e. `1.0.1`)
2. Releasing a new version (i.e. `2.0.0`) that either:
  a. Conforms to the updated usage of `Phoenix.Component`
  b. Includes `:phoenix_view` as a direct dependency so that `Phoenix.View.render_to_string/3` can still be called

IMO 2a is the path that should be taken but I'd appreciate a maintainer's input before heading down a particular path.

**Comment by kyleboe** [comment]
It is also worth noting that there is some good debate going on here: https://github.com/swoosh/phoenix_swoosh/issues/287

**Comment by icr4** [comment]
We have a working branch in our fork: https://github.com/populimited/bamboo_phoenix
Might be useful for a pull request

**Comment by kyleboe** [comment]
That's be awesome to see your changes get merged in to this repo! I saw you opened a PR yesterday. Is there anyone from thoughtbot that has write access to this repo to get a PR like that merged in?

CC: @cpytel

**Comment by chrism2671** [comment]
@icr4 I see the PR was closed; was that on purpose?

**Comment by icr4** [comment]
> @icr4 I see the PR was closed; was that on purpose?

Yes since it is not production ready yet, docs and backward compatibility should be further improved
